rpm-writer/bin/rpmcanon.rs

```diff
@@ -56,16 +56,62 @@ fn usage(success: bool) -> i32 {
     }
 }
 
+mod progress_reader {
+    pub enum ReportProgress {
+        No,
+        Yes,
+    }
+    use File;
+    pub struct ProgressReportingReader {
+        inner: File,
+        do_report: Option<u64>,
+    }
+    impl ProgressReportingReader {
+        pub fn new(inner: File, report: ReportProgress) -> Self {
+            match report {
+                ReportProgress::Yes => Self {
+                    inner,
+                    do_report: Some(0),
+                },
+                ReportProgress::No => Self {
+                    inner,
+                    do_report: None,
+                },
+            }
+        }
+    }
+    impl std::io::Read for ProgressReportingReader {
+        fn read(&mut self, buffer: &mut [u8]) -> std::io::Result<usize> {
+            let bytes_read = self.inner.read(buffer)?;
+            if let Some(ref mut bytes_so_far) = self.do_report {
+                if bytes_read == 0 {
+                    println!("{} bytes total", *bytes_so_far);
```

Comment:
IMO better use consistent units - either both bytes, or both MiB. In either case, reporting progress only every 1MiB sounds fine.